src/decoders/oem/test/oem_test.cpp

```diff
@@ -1027,9 +1027,9 @@ class DecodeEncodeTest : public ::testing::Test
         int32_t iRetCode = DecodeEncode(eFormat_, pstExpectedMessageData_->pucMessage, pucEncodeBuffer, MAX_ASCII_MESSAGE_LENGTH, stMetaData, stMessageData);
 
         return iRetCode != SUCCESS                                                                                   ? iRetCode
-               : eFormat_ != ENCODE_FORMAT::JSON && stMetaData.uiHeaderLength != stMessageData.uiMessageHeaderLength ? HEADER_LENGTH_ERROR
-               : pstExpectedMessageData_->uiMessageLength != stMessageData.uiMessageLength                           ? LENGTH_ERROR
-               : stMessageData != *pstExpectedMessageData_                                                           ? MESSAGE_DATA_COMPARISON_ERROR
+               : eFormat_ != ENCODE_FORMAT::JSON && stMetaData.uiHeaderLength != stMessageData.uiMessageHeaderLength ? HEADER_LENGTH_ERROR // NOLINT
+               : pstExpectedMessageData_->uiMessageLength != stMessageData.uiMessageLength                           ? LENGTH_ERROR // NOLINT
+               : stMessageData != *pstExpectedMessageData_                                                           ? MESSAGE_DATA_COMPARISON_ERROR // NOLINT
```

Comment:
why NOLINT?

Reply:
It's for `*-avoid-nested-conditional-operator` checks on newer clang-tidy versions. Should I disable it in `.clang-tidy` instead? I already converted all nested ternary operators to switches where possible in this PR.

Reply:
ahh I see. I think the change to using switches is for the best. I dont see .clang-tidy showing up in the diff though, so why is this NOLINT necessary now when it wasnt before?

Reply:
I'm assuming this is due to using clang-tidy 19 on my dev machine while the version in CI is 17. So the NOLINTs are currently not really needed as far as the CI is concerned.